driver/src/test/scala/ChangeStreamSpec.scala

```diff
@@ -0,0 +1,214 @@
+            event.getAs[BSONDocument]("fullDocument") must beSome(testDocument1)
+        }.await(retries = 2, 1.second)
+      }
+    }
+
+    "lookup the most recent document version" in skipIfNotRSAndNotVersionAtLeast(MongoWireVersion.V36) {
+      withTmpCollection(db) { coll: BSONCollection =>
+        // given
+        val cursor = coll.watch[BSONDocument]().cursor[Cursor.WithOps]
+        val id = "lookup_test1"
+        val fieldName = "foo"
+        val lastValue = "bar3"
+        val testDocument = BSONDocument(
+          "_id" -> id,
+          fieldName -> "bar1",
+        )
+
+        // This test is a bit more tricky. We want to first capture the insert event so that we know where we will
+        // resume. Then we produce two update events, resume the stream with the first update event, but check that the
+        // looked-up document corresponds to the second update.
+        val firstEventFuture = foldOne(cursor)
+        // See comment above
+        delayBy(500.millis) {
+          coll.insert(ordered = false).one(testDocument)
+        } must beEqualTo(()).await
+        val result = firstEventFuture.flatMap { firstEvent =>
+          firstEvent.get("_id") match {
+            case None => Future.failed(new Exception("The event had no id"))
+            case Some(eventId) =>
+
+              // when
+              for {
+                _ <- coll.update(ordered = false).one(
+                  BSONDocument("_id" -> id),
+                  BSONDocument(f"$$set" -> BSONDocument(fieldName -> "bar2"))
+                )
+                _ <- coll.update(ordered = false).one(
+                  BSONDocument("_id" -> id),
+                  BSONDocument(f"$$set" -> BSONDocument(fieldName -> lastValue))
+                )
+                event <- foldOne(
+                  coll.watch[BSONDocument](
+                    resumeAfter = Some(eventId),
+                    fullDocument = Some(ChangeStreams.FullDocument.UpdateLookup)
+                  ).cursor[Cursor.WithOps]
+                )
+              } yield event
+          }
+        }
+
+        // then
+        result must beLike[BSONDocument] {
+          case event =>
+            event.getAs[String]("operationType") must beSome("update")
+            event.getAs[BSONDocument]("documentKey")
+              .flatMap(_.getAs[String]("_id")) must beSome(id)
+            event.getAs[BSONDocument]("fullDocument")
+              .flatMap(_.getAs[String](fieldName)) must beSome(lastValue)
+        }.await(retries = 2, 1.second)
+      }
+    }
+  }
+
+  private def skipIfNotRSAndNotVersionAtLeast[R: AsResult](version: MongoWireVersion)(r: => R) = {
+    skippedIf(isNotReplicaSet, isNotAtLeast(db, version))(r)
+  }
+
+  private val actorSystem = ActorSystem("changeStreams")
+
+  override def afterAll: Unit = {
+    Await.ready(actorSystem.terminate(), timeout)
+    super.afterAll
+  }
+
+  private def foldOne[T](cursor: Cursor.WithOps[T]): Future[T] = {
```

Comment:
`cursor.head` ?

Reply:
precisely not :) `headOption` will alway return `None` on a change stream cursor

Reply:
`cursor.head` will send a single `Aggregate` command which will always return no elements in this case, unless the stream is being resumed.

`cursor.collect(maxDocs = 1)` will send the initial `Aggregate` command, and then will continue sending `GetMore` commands until a new event is available.